lib/api/programsLogic.ts

```diff
@@ -0,0 +1,46 @@
+import prisma from '../../prisma/prisma';
+import { ProgramsWrite, programsWriteSchema } from '../types/dtos/createProgramsRequest.dto';
+import { BadRequestError } from '../utils/errors/badRequestError';
+import type { NextApiRequest, NextApiResponse } from 'next';
+import { S3_BASE_URL, uploadToS3 } from '../aws';
+import { getImageBufferFromBase64 } from '../utils/utils';
+
+export const getAllProgramsHandler = async (req: NextApiRequest, res: NextApiResponse) => {
+  res.status(200).json(await prisma.programs.findMany());
+};
+
+export const postProgramHandler = async (req: NextApiRequest, res: NextApiResponse) => {
+  const {
+    title,
+    content,
+    base64EncodedImage,
+    eventDateISOString: eventDate,
+    location,
+  }: ProgramsWrite = validateCreateProgramRequest(req.body);
+
+  const eventDateWithoutPeriods: string = eventDate.replace('.', '_');
+  await uploadToS3(
```

Comment:
Ah, I see, wouldnt the underscore replacement be more appropriate in this function instead?

Reply:
which function are you talking about?